File: onderwijs4u-back-end/src/main/java/nl/han/devops/service/AuthenticationService.java
package nl.han.devops.service;

import nl.han.devops.data.AuthenticationDAO;
import org.springframework.stereotype.Service;

@Service
public class AuthenticationService {
    private final AuthenticationDAO authenticationDAO;

    public AuthenticationService(AuthenticationDAO authenticationDAO) {
        this.authenticationDAO = authenticationDAO;
    }

    public boolean tokenExists(String token) {
        return authenticationDAO.tokenExists(token);
    }
}

File: back-end/src/main/kotlin/nl/lpp/uncenshow/service/authentication/AuthenticationService.kt
package nl.lpp.uncenshow.service.authentication

import nl.lpp.uncenshow.datasource.authentication.AuthenticationDAO
import org.springframework.stereotype.Service

@Service
class AuthenticationService (private val authenticationDAO: AuthenticationDAO) {

    fun tokenExists(token: String): Boolean {
        return authenticationDAO.tokenExists(token)
    }
}